jwst/tweakreg/tweakreg_step.py

```diff
@@ -200,13 +202,94 @@ def process(self, input):
+            if self.save_gaia_catalog:
+                output_name = 'fit_{}_ref.ecsv'.format(self.gaia_catalog.lower())
+            else:
+                output_name = None
+            ref_cat = amutils.create_astrometric_catalog(images,
+                                                         self.gaia_catalog,
+                                                         output=output_name)
+
+            # Check that there are enough GAIA sources for a reliable/valid fit
+            num_ref = len(ref_cat)
+            if num_ref < self.min_gaia:
+                msg = "Not enough GAIA sources for a fit: {}\n".format(num_ref)
+                msg += "Skipping alignment to {} astrometric catalog!\n".format(self.gaia_catalog)
+                # Raise Exception here to avoid rest of code in this try block
+                self.log.warning(msg)
+            else:
+                # align images:
+                # Update to separation needed to prevent confusion of sources 
+                # from overlapping images where centering is not consistent or
+                # for the possibility that errors still exist in relative overlap.  
+                tpmatch_gaia = TPMatch(
+                    searchrad=self.searchrad,
+                    separation=self.separation / 5.0,
+                    use2dhist=self.use2dhist,
+                    tolerance=self.tolerance,
+                    xoffset=0.0,
+                    yoffset=0.0
+                )
+
+                # Set group_id to same value so all get fit as one observation
+                # The assigned value, 987654, has been hard-coded to make it
+                # easy to recognize when alignment to GAIA was being performed
+                # as opposed to the group_id values used for relative alignment
+                # earlier in this step.
+                for imcat in imcats:
+                    imcat.meta['orig_group_id'] = imcat.meta['group_id']
+                    imcat.meta['group_id'] = 987654
+                    if 'REFERENCE' in imcat.meta['fit_info']['status']:
+                        del imcat.meta['fit_info']
+                    
+                # Perform fit
+                align_wcs(imcats,
+                          refcat=ref_cat,
+                          enforce_user_order=False,
+                          expand_refcat=False,
+                          minobj=self.minobj,
+                          match=tpmatch_gaia,
+                          fitgeom=self.fitgeometry,
+                          nclip=self.nclip,
+                          sigma=(self.sigma, 'rmse')
+                         )
+
+                # Reset group_id to original values
+                # Also, update/create the WCS .name attribute with information 
+                #  on this astrometric fit as the only record that it was successful
+                for imcat in imcats:
+                    imcat.meta['group_id'] = imcat.meta['orig_group_id']
```

Comment:
I think this `'group_id'` is not used anywhere in the pipeline or stored/recorded in the `DataModel` and so there is no need to "restore" this id.